core/geometry/src/curve/LineString3d.ts

```diff
@@ -931,6 +931,31 @@ public override rangeBetweenFractions(fraction0: number, fraction1: number, tran
 
   }
 
+  /** Compress out duplicate points (according to point.isAlmostEqual)
+   */
+   public removeDuplicatePoints(tolerance: number = Geometry.smallMetricDistance) {
+    const n = this._points.length;
+    if (n < 2)
+      return;
+    let n1 = 1;
+    for (let i = 1; i < n; i++){
+      const q = this._points.distanceIndexIndex (i, n1 - 1);
+      if (q !== undefined && q > tolerance){
+        this._points.moveIndexToIndex (i, n1);
+        if (this._fractions !== undefined)
+          this._fractions.setAtUncheckedIndex (i, this._fractions.atUncheckedIndex(n1));
```

Comment:
I think i and n1 should be swapped so that we overwrite the fraction at n1 like we do for points and derivatives.

Reply:
Corrected in push [13b8e1e](https://github.com/iTwin/itwinjs-core/pull/4194/commits/13b8e1e6cc7bfaac8dde89df66053e82baa37e95)